src/v/resource_mgmt/storage.h

```diff
@@ -243,6 +243,116 @@ class disk_space_manager {
+         * management. This is generally a bit different from the total amount
+         * of space that redpanda is using because space management only
+         * considers data that it may eventually be able to reclaim. This value
+         * should reflect the total disk usage for both cloud and non-cloud
+         * enabled topics.
+         */
+        void set_total_usage(size_t usage) noexcept { _total_usage = usage; }
+
+        /*
+         * This is the amount of data that is currently reclaimable by the
+         * normal retention policy. Space management only kicks in when it needs
+         * to reclaim more data than is available through the normal retention
+         * mechanism.
+         */
+        void set_retention_reclaimable(size_t size) noexcept {
+            _retention_reclaimable = size;
+        }
+
+        /*
+         * The total available amount for reclaim by space management is the
+         * amount of data reclaimable by normal retention policy plus the amount
+         * of data has been uploaded into the cloud, and thus safe to remove if
+         * necessary to meet the configured usage target.
+         */
+        void set_available_reclaimable(size_t size) noexcept {
+            _available_reclaimable = size;
+        }
+
+        /*
+         * The total amount of data that can be safely reclaimed and which if
+         * reclaimed would not violate the local retention policy.
+         */
+        void set_local_retention_reclaimable(size_t size) noexcept {
+            _local_retention_reclaimable = size;
+        }
+
+        /*
+         * The amount of data usage above the target threshold. when space
+         * management runs if this value is positive then space management will
+         * try to ensure this much data is reclaimed, if that is by applying
+         * normal retention policy or applying space management over policy.
+         */
+        void set_target_excess(size_t size) noexcept { _target_excess = size; }
+
+        /*
+         * The amount of data estimated to be reclaimed by the currently
+         * installed space management schedule that is above the local retention
+         * policy setting.
+         */
+        void set_reclaim_local(size_t size) noexcept { _reclaim_local = size; }
+
+        /*
+         * The amount of data estimated to be reclaimed by the currently
+         * installed space management schedule that is above the non-hinted
+         * low-space threshold setting.
+         */
+        void set_reclaim_low_non_hinted(size_t size) noexcept {
+            _reclaim_low_non_hinted = size;
+        }
+
+        /*
+         * The amount of data estimated to be reclaimed by the currently
+         * installed space management schedule that is above the hinted
+         * low-space threshold setting.
+         */
+        void set_reclaim_low_hinted(size_t size) noexcept {
+            _reclaim_low_hinted = size;
+        }
```

Comment:
I imagine the "hinted" vs "non-hinted" might be a bit confusing to explain. Are we try to expose these details? Is that the intention for making these non-public metrics?

Reply:
Actually it seems these aren't exposed at all -- is that intentional?

Reply:
1. aren't exposed: this was either an oversight or a git rebase mistake. they are now exposed as intended.
2. hinted/non-hinted: definitely want to make these low-level bits non-public. the goal here is to empower support, and what i've heard occasionally is an interest in how much data qualifies for each "category" of reclaimable data without having to turn on logging.